core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      const { getByTestId, container } = render(<Input name="test" icon="search" error={true} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      const leftIcon = container.querySelector('.Input-icon--left');
+
+      fireEvent.focus(input);
+      expect(wrapper).toHaveClass('Input--error');
+      expect(leftIcon).toBeInTheDocument();
+    });
+
+    it('applies disabled state CSS correctly', () => {
+      const { getByTestId, container } = render(<Input name="test" icon="search" disabled={true} />);
+
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      const leftIcon = container.querySelector('.Input-icon--left');
+
+      expect(wrapper).toHaveClass('Input--disabled');
+      expect(leftIcon).toBeInTheDocument();
+    });
+
+    it('applies readOnly state CSS correctly', () => {
+      const { getByTestId } = render(<Input name="test" readOnly={true} />);
+
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      expect(wrapper).toHaveClass('Input--readOnly');
+    });
+
+    it('applies tiny size icon styling correctly using utility classes', () => {
+      const { getByTestId, container } = render(<Input name="test" size="tiny" icon="search" />);
+
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      const leftIcon = container.querySelector('.Input-icon--left');
+
+      expect(wrapper).toHaveClass('Input--tiny');
+      expect(leftIcon).toBeInTheDocument();
+      expect(leftIcon).toHaveClass('mr-3');
+    });
+  });
+
+  describe('Icon Type and States', () => {
+    it('renders icon with specified iconType', () => {
+      const { container } = render(<Input name="test" icon="search" iconType="outlined" value="test" />);
+
+      const leftIcon = container.querySelector('.Input-icon--left');
+      expect(leftIcon).toBeInTheDocument();
+    });
+
+    it('handles input blank state for icon styling', () => {
+      const { getByTestId, container } = render(<Input name="test" icon="search" />);
+
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      const leftIcon = container.querySelector('.Input-icon--left');
+
+      expect(leftIcon).toBeInTheDocument();
+
+      fireEvent.change(input, { target: { value: 'test' } });
+      fireEvent.blur(wrapper);
+      expect(leftIcon).toBeInTheDocument();
+
+      fireEvent.change(input, { target: { value: '' } });
+      fireEvent.blur(wrapper);
+      expect(leftIcon).toBeInTheDocument();
+    });
+
+    it('applies error state to icon correctly', () => {
+      const { container } = render(<Input name="test" icon="search" error={true} value="test" />);
+
+      const leftIcon = container.querySelector('.Input-icon--left');
+      expect(leftIcon).toBeInTheDocument();
+    });
+  });
+
+  describe('Tiny Input Icon Support', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.